File: python/design-patterns/composite/composite/directory_composite.py
class DirectoryComposite(FileSystemNode):
    def __init__(self, name: str):
        self.name = name
        self.children = []
    def get_name(self): return self.name
    def add(self, node): self.children.append(node)
    def get_size(self):
        return sum(child.get_size() for child in self.children)

File: cpp/design-patterns/composite/composite/DirectoryComposite.cpp
class DirectoryComposite : public FileSystemNode {
    std::string name;
    std::vector<FileSystemNode*> children;
public:
    DirectoryComposite(std::string n) : name(n) {}
    std::string getName() override { return name; }
    void add(FileSystemNode* node) { children.push_back(node); }
    int getSize() override {
        int sum = 0;
        for (auto child : children) {
            sum += child->getSize();
        }
        return sum;
    }
};